guava/src/com/google/common/hash/BloomFilter.java

```diff
@@ -599,19 +601,22 @@ public void writeTo(OutputStream out) throws IOException {
     int dataLength = -1;
     try {
       DataInputStream din = new DataInputStream(in);
-      // currently this assumes there is no negative ordinal; will have to be updated if we
+      // currently, this assumes there is no negative ordinal; will have to be updated if we
       // add non-stateless strategies (for which we've reserved negative ordinals; see
       // Strategy.ordinal()).
       strategyOrdinal = din.readByte();
       numHashFunctions = UnsignedBytes.toInt(din.readByte());
       dataLength = din.readInt();
 
       Strategy strategy = BloomFilterStrategies.values()[strategyOrdinal];
-      long[] data = new long[dataLength];
-      for (int i = 0; i < data.length; i++) {
-        data[i] = din.readLong();
+
+      LockFreeBitArray lockFreeBitArray = new LockFreeBitArray(LongMath.checkedMultiply(dataLength,
+          64L));
+      for (int i = 0; i < dataLength; i++) {
+        lockFreeBitArray.data.set(i, din.readLong());
```

Comment:
If you had used the data object directly, I see https://github.com/google/guava/blob/master/guava/src/com/google/common/hash/BloomFilterStrategies.java#L179 has a AtomicLongArray, does this now change the fact that its not thread safe anymore?

Reply:
The constructor with size of the array as param initiates an AtomicLongArray. So nothing changes.
https://github.com/google/guava/blob/e64ad2634216a3bb8e2beeb26d673f72d24e293a/guava/src/com/google/common/hash/BloomFilterStrategies.java#L172